Content.Server/Supermatter/EntitySystems/SupermatterSystem.cs

```diff
@@ -0,0 +1,425 @@
+                    }
+
+                    //if there are space tiles next to SM
+                    foreach(var adjacent in _atmosphereSystem.GetAdjacentTileMixtures(Xform.Coordinates))
+                    {
+                        if (adjacent.TotalMoles == 0)
+                        {
+                            if (integrity < 90)
+                                damage = Math.Clamp((SMcomponent.Power * 0.0005f) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                            else if (integrity < 75)
+                                damage = Math.Clamp((SMcomponent.Power * 0.0009f) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                            else if (integrity < 55)
+                                damage = Math.Clamp((SMcomponent.Power * 0.005f) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                            else if (integrity < 25)
+                                damage = Math.Clamp((SMcomponent.Power * 0.002f) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                            else
+                                break;
+                        }
+                    }
+                }
+
+                //only take up to 1.8 damage per cycle with no lower limmit
+                damage = Math.Min(SMcomponent.DamageArchived + (SupermatterComponent.DamageHardcap * SupermatterComponent.ExplosionPoint), damage);
+                //damage to add to total
+                var damageDelta = new DamageSpecifier(_prototypeManager.Index<DamageTypePrototype>("Blunt"), (Shared.FixedPoint.FixedPoint2)damage);
+                _damageable.TryChangeDamage(SMcomponent.Owner, damageDelta, true);
+            }
+        }
+
+        /// </summary>
+        /// Runs the logic and timers for Delamination
+        /// </summary>
+        public void Delamination(EntityUid Uid, float frameTime, SupermatterComponent? SMcomponent = null, TransformComponent? Xform = null)
+        {
+            if (!Resolve(Uid, ref SMcomponent, ref Xform))
+            {
+                return;
+            }
+
+            //TODO: make tesla spawn at SupermatterComponent.PowerPenaltyThreshold
+            if (!SMcomponent.FinalCountdown)
+            {
+                if (_atmosphereSystem.GetTileMixture(Xform.Coordinates) is { } mixture)
+                {
+                    if (mixture.TotalMoles >= SupermatterComponent.MolePenaltyThreshold)
+                    {
+                        _chatManager.DispatchStationAnnouncement(Loc.GetString("supermatter-delamination-overmass"), "Supermatter", false);
+                    }
+                    else
+                    {
+                        _chatManager.DispatchStationAnnouncement(Loc.GetString("supermatter-delamination-default"), "Supermatter", false);
```

Comment:
the "Supermatter" isn't localized anyway and should just pull from the entity name as I mentioned before